Source/Charts/Utils/ChartUtils.swift

```diff
@@ -156,96 +166,96 @@ open class ChartUtils
     {
         var drawOffset = CGPoint()
-        
-        NSUIGraphicsPushContext(context)
-        
+
+        NSUIGraphicsPushContext(self)
+
         if angleRadians != 0.0
         {
             let size = text.size(withAttributes: attributes)
-            
+
             // Move the text drawing rect in a way that it always rotates around its center
             drawOffset.x = -size.width * 0.5
             drawOffset.y = -size.height * 0.5
-            
+
             var translate = point
-            
+
             // Move the "outer" rect relative to the anchor, assuming its centered
             if anchor.x != 0.5 || anchor.y != 0.5
             {
                 let rotatedSize = size.rotatedBy(radians: angleRadians)
-                
+
                 translate.x -= rotatedSize.width * (anchor.x - 0.5)
                 translate.y -= rotatedSize.height * (anchor.y - 0.5)
             }
-            
-            context.saveGState()
-            context.translateBy(x: translate.x, y: translate.y)
-            context.rotate(by: angleRadians)
-            
+
+            saveGState()
+            translateBy(x: translate.x, y: translate.y)
+            rotate(by: angleRadians)
+
             (text as NSString).draw(at: drawOffset, withAttributes: attributes)
-            
-            context.restoreGState()
+
+            restoreGState()
         }
         else
         {
             if anchor.x != 0.0 || anchor.y != 0.0
             {
                 let size = text.size(withAttributes: attributes)
-                
+
                 drawOffset.x = -size.width * anchor.x
                 drawOffset.y = -size.height * anchor.y
             }
-            
+
             drawOffset.x += point.x
             drawOffset.y += point.y
-            
+
             (text as NSString).draw(at: drawOffset, withAttributes: attributes)
         }
-        
+
         NSUIGraphicsPopContext()
     }
-    
-    internal class func drawMultilineText(context: CGContext, text: String, knownTextSize: CGSize, point: CGPoint, attributes: [NSAttributedStringKey : Any]?, constrainedToSize: CGSize, anchor: CGPoint, angleRadians: CGFloat)
+
+    internal func drawMultilineText(_ text: String, at point: CGPoint, constrainedTo size: CGSize, anchor: CGPoint, knownTextSize: CGSize, angleRadians: CGFloat, attributes: [NSAttributedStringKey : Any]?)
     {
-        var rect = CGRect(origin: CGPoint(), size: knownTextSize)
-        
-        NSUIGraphicsPushContext(context)
-        
+        var rect = CGRect(origin: .zero, size: knownTextSize)
+
+        NSUIGraphicsPushContext(self)
+
         if angleRadians != 0.0
         {
             // Move the text drawing rect in a way that it always rotates around its center
             rect.origin.x = -knownTextSize.width * 0.5
             rect.origin.y = -knownTextSize.height * 0.5
-            
+
```

Comment:
any chance this is something wrong with your setup? About deleting some spaces? You intend to do this or ?

Reply:
I have remove white space in white space only lines turned on. No point in having a line of empty space.